engine/src/main/java/com/redhat/cloud/notifications/events/FromOpenBridgeHistoryFiller.java

```diff
@@ -0,0 +1,133 @@
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.quarkus.scheduler.Scheduled.ConcurrentExecution.SKIP;
+
+/**
+ *
+ */
+@ApplicationScoped
+public class FromOpenBridgeHistoryFiller {
+
+    public static final String ORIGINAL_EVENT_ID_HEADER = "rhose-original-event-id";
+    public static final String MESSAGES_WITH_ERROR_NAME = "rhose.messages.error";
+    public static final String DEAD_LETTER_CAUSE = "dead-letter-cause";
+
+    @ConfigProperty(name = "ob.backchannel-filler.enabled", defaultValue = "true")
+    boolean pollEnabled;
+
+    @Inject
+    @RestClient
+    BridgeApiService bridgeApiService;
+
+    @Inject
+    Bridge bridge;
+
+    @Inject
+    BridgeAuth bridgeAuth;
+
+    @Inject
+    NotificationHistoryRepository notificationHistoryRepository;
+
+    @Inject
+    StatelessSessionFactory statelessSessionFactory;
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    private Counter messagesWithError;
+
+    private List<String> seenIds; // TODO should be some kind of real cache with expiry
+
+    @PostConstruct
+    void init() {
+        messagesWithError = meterRegistry.counter(MESSAGES_WITH_ERROR_NAME);
+        seenIds = new ArrayList<>();
+    }
+
+    @Scheduled(concurrentExecution = SKIP, every = "${ob.error-check.period:10s}")
+    @Transactional
+    public void execute() {
+
+        if (!pollEnabled) {
+            return;
+        }
+
+        BridgeItemList<ProcessingError> errorList = bridgeApiService.getProcessingErrors(bridge.getId(), bridgeAuth.getToken());
+
+        // OB offers us a list of items that represent the last X errors
+        // We may have seen individual items before
+        List<ProcessingError> items = errorList.getItems();
+        if (items == null) {
+            return;
+        }
+
+        for (ProcessingError pe : items) {
+
+            // We may have seen an item before, skip it
+//            String historyId = pe.getHeaders().get(ORIGINAL_EVENT_ID_HEADER);
+//            if (alreadyProcessed(historyId)) {
+//                continue;
+//            }
+
+            statelessSessionFactory.withSession(statelessSession -> {
+                Map<String, Object> decodedPayload = decodeItem(pe);
+                // TODO reinjectIf needed (?)
+                try {
+                    notificationHistoryRepository.updateHistoryItem(decodedPayload);
+                    messagesWithError.increment();
+                } catch (Exception e) {
+                    Log.info("|  Update Fail", e);
+                }
+            });
+        }
+    }
+
+    private Map<String, Object> decodeItem(ProcessingError pe) {
+        Map<String, Object> map = new HashMap<>();
+        Map<String, String> headers = pe.getHeaders();
+        map.put("historyId", headers.get(ORIGINAL_EVENT_ID_HEADER));
```

Comment:
This small change would make it obvious that we're talking about a Smart Events `event` and not a Notifications `event`:
```suggestion
        map.put("historyId", headers.get(ORIGINAL_SMART_EVENT_ID_HEADER));
```

Reply:
Well technically it is a header from SE, but the content is the original ID we submitted from the Notifications side